diff --git a/printing/BUILD.gn b/printing/BUILD.gn
--- a/printing/BUILD.gn
+++ b/printing/BUILD.gn
@@ -16,5 +16,3 @@
 if (enable_printing_tests) {
   import("//build/config/ios/bundle_data_from_filelist.gni")
-} else {
-  assert(enable_printing)
 }
diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1208,1 +1208,0 @@
-      "//chrome/browser/printing",
@@ -1506,6 +1505,7 @@
 
   if (enable_printing) {
     deps += [
+      "//chrome/browser/printing",
       "//chrome/browser/printing:impl",
       "//chrome/common/printing",
       "//components/printing/browser/headless",
diff --git a/chrome/browser/ui/webui/print_preview/BUILD.gn b/chrome/browser/ui/webui/print_preview/BUILD.gn
--- a/chrome/browser/ui/webui/print_preview/BUILD.gn
+++ b/chrome/browser/ui/webui/print_preview/BUILD.gn
@@ -8,6 +8,6 @@
 import("//pdf/features.gni")
 import("//printing/buildflags/buildflags.gni")
 
-assert(enable_print_preview)
+assert(enable_print_preview || !enable_printing)
 
 # Header-only target exposing the print_preview public API. This is kept
diff --git a/components/printing/common/BUILD.gn b/components/printing/common/BUILD.gn
--- a/components/printing/common/BUILD.gn
+++ b/components/printing/common/BUILD.gn
@@ -6,6 +6,6 @@
 import("//mojo/public/tools/bindings/mojom.gni")
 import("//printing/buildflags/buildflags.gni")
 
-assert(enable_printing)
+assert(enable_printing || !enable_print_preview)
 
 if (enable_print_media_l10n) {
diff --git a/chrome/browser/ui/webui/print_preview/BUILD.gn b/chrome/browser/ui/webui/print_preview/BUILD.gn
--- a/chrome/browser/ui/webui/print_preview/BUILD.gn
+++ b/chrome/browser/ui/webui/print_preview/BUILD.gn
@@ -165,6 +165,6 @@
       "//components/user_manager",
     ]
-  } else {
+  } else if (enable_print_preview) {
     sources += [ "local_printer_handler_default.cc" ]
     deps += [ "//chrome/browser/resources/print_preview:resources" ]
   }
