diff --git a/content/browser/devtools/features.gni b/content/browser/devtools/features.gni
--- a/content/browser/devtools/features.gni
+++ b/content/browser/devtools/features.gni
@@ -8,1 +8,1 @@
-enable_devtools_frontend = !is_ios && !is_fuchsia
+enable_devtools_frontend = false
@@ -9,0 +10,1 @@
+enable_devtools_browser = true
diff --git a/chrome/browser/devtools/BUILD.gn b/chrome/browser/devtools/BUILD.gn
--- a/chrome/browser/devtools/BUILD.gn
+++ b/chrome/browser/devtools/BUILD.gn
@@ -158,1 +158,1 @@
-  if (enable_devtools_frontend) {
+  if (enable_devtools_frontend || enable_devtools_browser) {
@@ -207,1 +207,1 @@
-    if (toolkit_views) {
+    if (defined(toolkit_views) && toolkit_views) {
