--- a/printing/common/metafile_utils.cc
+++ b/printing/common/metafile_utils.cc
@@ -40,6 +40,8 @@
 
 namespace {
 
+#if BUILDFLAG(ENABLE_PRINTING)
+
 SkString GetHeadingStructureType(int heading_level) {
   // From Table 366 in PDF 32000-2:2020 spec, section 14.8.4.5,
   // "H1"..."H6" are valid structure types.
@@ -64,6 +66,8 @@
       .fSecond = static_cast<uint8_t>(exploded.second)};
 }
 
+#endif  // BUILDFLAG(ENABLE_PRINTING)
+
 sk_sp<SkPicture> GetEmptyPicture() {
   SkPictureRecorder rec;
   SkCanvas* canvas = rec.beginRecording(100, 100);
@@ -72,6 +76,8 @@
   canvas->restore();
   return rec.finishRecordingAsPicture();
 }
+
+#if BUILDFLAG(ENABLE_PRINTING)
 
 void AppendCheckedStateIfTrue(const ui::AXNode* ax_node,
                               SkPDF::StructureElementNode* tag) {
@@ -387,6 +393,8 @@
   return valid;
 }
 
+#endif  // BUILDFLAG(ENABLE_PRINTING)
+
 sk_sp<const SkData> GetImageData(SkImage* img) {
   // Skip the encoding step if the image is already encoded
   if (auto data = img->refEncodedData()) {
@@ -412,6 +420,7 @@
     const ui::AXTreeUpdate& accessibility_tree,
     mojom::GenerateDocumentOutline generate_document_outline,
     SkWStream* stream) {
+#if BUILDFLAG(ENABLE_PRINTING)
   SkPDF::Metadata metadata;
   SkPDF::DateTime now = TimeToSkTime(base::Time::Now());
   metadata.fCreation = now;
@@ -439,6 +448,9 @@
   }
 
   return SkPDF::MakeDocument(stream, metadata);
+#else
+  return nullptr;
+#endif  // BUILDFLAG(ENABLE_PRINTING)
 }
 
 SkSerialReturnType SerializeOopPicture(SkPicture* pic, void* ctx) {
