diff --git a/chrome/app/chrome_main_delegate.cc b/chrome/app/chrome_main_delegate.cc
--- a/chrome/app/chrome_main_delegate.cc
+++ b/chrome/app/chrome_main_delegate.cc
@@ -1627,3 +1627,4 @@ void ChromeMainDelegate::ZygoteForked() {
-  crash_reporter::InitializeCrashpad(false, process_type);
-  crash_reporter::SetFirstChanceExceptionHandler(
-      v8::TryHandleWebAssemblyTrapPosix);
+  if (crash_reporter::InitializeCrashpad(false, process_type)) {
+    crash_reporter::SetFirstChanceExceptionHandler(
+        v8::TryHandleWebAssemblyTrapPosix);
+  }
