diff --git a/third_party/dawn/src/dawn/native/SystemEvent.cpp b/third_party/dawn/src/dawn/native/SystemEvent.cpp
--- a/third_party/dawn/src/dawn/native/SystemEvent.cpp
+++ b/third_party/dawn/src/dawn/native/SystemEvent.cpp
@@ -38,7 +38,11 @@
 #include <poll.h>
 #include <unistd.h>
 #elif DAWN_PLATFORM_IS(POSIX)
+#if defined(__linux__) && !defined(__GLIBC__)
+#include <poll.h>
+#else
 #include <sys/poll.h>
+#endif
 #include <unistd.h>
 #endif
