diff --git a/chrome/browser/ui/views/extensions/extensions_toolbar_desktop.cc b/chrome/browser/ui/views/extensions/extensions_toolbar_desktop.cc
--- a/chrome/browser/ui/views/extensions/extensions_toolbar_desktop.cc
+++ b/chrome/browser/ui/views/extensions/extensions_toolbar_desktop.cc
@@ -476,7 +476,7 @@ void ExtensionsToolbarDesktop::AnchorAndShowWidgetImmediately(
   // and extension-uninstall-dialog may be out of the window border on Linux.
   if (base::CommandLine::ForCurrentProcess()->HasSwitch("hide-extensions-menu")) {
-    views::View* anchor_view =
+    views::View* menu_anchor_view =
         views::ElementTrackerViews::GetInstance()->GetFirstMatchingView(
             kToolbarAppMenuButtonElementId,
             views::ElementTrackerViews::GetContextForView(
                 BrowserView::GetBrowserViewForBrowser(browser_)));
     widget->widget_delegate()->AsBubbleDialogDelegate()
-      ->SetAnchorView(anchor_view);
+      ->SetAnchorView(menu_anchor_view);
   }
