diff --git a/base/allocator/partition_allocator/src/partition_alloc/pointers/raw_ref.h b/base/allocator/partition_allocator/src/partition_alloc/pointers/raw_ref.h
--- a/base/allocator/partition_allocator/src/partition_alloc/pointers/raw_ref.h
+++ b/base/allocator/partition_allocator/src/partition_alloc/pointers/raw_ref.h
@@ -446,0 +446,5 @@ struct less<raw_ref<T, Traits>> {
+  bool operator()(const T& lhs, const raw_ref<T, Traits>& rhs) const
+      requires (!std::is_const_v<T>) {
+    Impl::IncrementLessCountForTest();
+    return lhs < rhs;
+  }
@@ -450,0 +450,6 @@ struct less<raw_ref<T, Traits>> {
+
+  bool operator()(const raw_ref<T, Traits>& lhs, const T& rhs) const
+      requires (!std::is_const_v<T>) {
+    Impl::IncrementLessCountForTest();
+    return lhs < rhs;
+  }
