diff --git a/ui/gl/gl_context_egl.cc b/ui/gl/gl_context_egl.cc
--- a/ui/gl/gl_context_egl.cc
+++ b/ui/gl/gl_context_egl.cc
@@ -275,7 +275,10 @@
+  const bool global_semaphore_share_group =
+      attribs.global_semaphore_share_group &&
+      gl_display_->ext->b_EGL_ANGLE_display_semaphore_share_group;
   if (gl_display_->ext->b_EGL_ANGLE_display_semaphore_share_group) {
     context_attributes.push_back(EGL_DISPLAY_SEMAPHORE_SHARE_GROUP_ANGLE);
     context_attributes.push_back(
-        attribs.global_semaphore_share_group ? EGL_TRUE : EGL_FALSE);
+        global_semaphore_share_group ? EGL_TRUE : EGL_FALSE);
   } else {
-    DCHECK(!attribs.global_semaphore_share_group);
+    DCHECK(!global_semaphore_share_group);
   }
