diff --git a/ui/gl/gl_context_egl.cc b/ui/gl/gl_context_egl.cc
--- a/ui/gl/gl_context_egl.cc
+++ b/ui/gl/gl_context_egl.cc
@@ -264,1 +264,3 @@
-  global_texture_share_group_ = attribs.global_texture_share_group;
+  global_texture_share_group_ =
+      attribs.global_texture_share_group &&
+      gl_display_->ext->b_EGL_ANGLE_display_texture_share_group;
   if (gl_display_->ext->b_EGL_ANGLE_display_texture_share_group) {
     context_attributes.push_back(EGL_DISPLAY_TEXTURE_SHARE_GROUP_ANGLE);
     context_attributes.push_back(global_texture_share_group_ ? EGL_TRUE
