--- a/net/socket/udp_socket_posix.cc
+++ b/net/socket/udp_socket_posix.cc
@@ -1089,6 +1089,10 @@
     return;
   }
 
+#if defined(__linux__) && !defined(__GLIBC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-compare"
+#endif
   for (struct cmsghdr* cmsg = CMSG_FIRSTHDR(msg);
        cmsg != nullptr &&
        (!result->tos.has_value() ||
@@ -1144,6 +1148,9 @@
     }
 #endif
   }
+#if defined(__linux__) && !defined(__GLIBC__)
+#pragma GCC diagnostic pop
+#endif
 }
 
 base::expected<DatagramsMetadata, Error> UDPSocketPosix::InternalReadMultiple(
